+bids/+internal/parse_filename.m

```diff
@@ -78,39 +76,81 @@
   end
   basename = filename(pos:end);
 
+  % removing extention
+  [basename, p.ext] = strtok(basename, '.');
+
   % -Identify all the BIDS entity-label pairs present in the filename (delimited by "_")
   [parts, dummy] = regexp(basename, '(?:_)+', 'split', 'match'); %#ok<ASGLU>
   p.filename = filename;
 
   % -Identify the suffix and extension of this file
-  [p.suffix, p.ext] = strtok(parts{end}, '.');
+  p.entities = struct();
+  p.suffix = '';
 
   % -Separate the entity from the label for each pair identified above
-  for i = 1:numel(parts) - 1
+  for i = 1:numel(parts)
+
     try
-      [d, dummy] = regexp(parts{i}, '(?:\-)+', 'split', 'match'); %#ok<ASGLU>
-      p.entities.(d{1}) = d{2};
-    catch
-      msg = sprintf(['Entity-label pair %s of file %s is not valid.\n', ...
-                     'This could also be to a suffix with an extra _'], parts{i}, filename);
+
+      if isempty(parts{i})
```

Comment:
am struggling to come up with a test case that hits this. Can you think of one?

Reply:
FYI I am ok to leave it in but I might add a comment to mention that this is unlikely to ever happen.

Reply:
Just to cover weird cases with double '_':
```matlab
[parts, dummy] = regexp('_a_', '(?:_)+', 'split', 'match')

parts =

  1x3 cell array

    {0x0 char}    {'a'}    {0x0 char}
```

Reply:
Actually, it just for cases of starting and ending with '_', it is nt catching duble underscore:
```matlab
parts, dummy] = regexp('a__b', '(?:_)+', 'split', 'match')

parts =

  1x2 cell array

    {'a'}    {'b'}


dummy =

  1x1 cell array

    {'__'}
```

Reply:
Cool!!! 

Added a test to cover that error too.